src/server/worker/worker.go

```diff
@@ -60,9 +60,14 @@ func NewWorker(
 	}
 
 	if pipelineInfo.Details.Transform.Image != "" && pipelineInfo.Details.Transform.Cmd == nil {
-		ppsutil.FailPipeline(env.Context(), env.GetDBClient(), driver.Pipelines(),
+		if err := ppsutil.FailPipeline(
+			env.Context(),
+			env.GetDBClient(),
+			driver.Pipelines(),
 			pipelineInfo.SpecCommit,
-			"nothing to run: no transform.cmd")
+			"nothing to run: no transform.cmd"); err != nil {
+			return nil, err
+		}
```

Comment:
this should probably have been returning regardless, the workers for the failed pipeline will get cleaned up, anyway